server/helpers/dateHelpers.js

```diff
@@ -46,3 +46,63 @@ export function isValidStartDate({ endDate = moment(), startDate = moment().subt
   // check if both date from and to are provided
   return { dateFrom, dateTo };
 }
+
+/**
+ * helper for the date Query
+ *
+ * @param {object} duration - duration parameter
+ * @param {boolean} trends - trends parameter
+ *
+ * @returns {number} response representing the date value
+ */
+const dateSubtractor = (duration, trends) => {
+  let date;
+  if (duration === 'days') {
+    switch (trends) {
+      case true:
+        date = 30;
+        return date;
+      default:
+        date = 0;
+        return date;
+    }
+  }
+  date = 1;
+  return date;
+};
+
+/**
+ * helper for the duration check
+ *
+ * @param {object} duration - duration parameter
+ *
+ * @returns {object} response representing the error
+ */
+export const checkDuration = (duration) => {
+  if (duration !== 'days' && duration !== 'weeks' && duration !== 'months' && duration !== 'years') {
+    const error = 'invalid duration input';
+    return error;
+  }
+  return false;
+};
+
+/**
+ * helper for selecting data of a specific time period
+ *
+ * @param {object} req - REST request object
+ * @param {boolean} trends - Trends parameter
+ * @param {number} duration - Time query parameter
+ *
+ * @returns {object} response reporting the start and end dates respectively
+ */
+export const dateQuery = (req, trends, duration) => {
```

Comment:
To make this method cleaner (purer), pass the date from the `req` object directly rather than the HTTP req object itself

Reply:
resolved